### PR TITLE
Ignoring empty query string parameters sent to search endpoint

### PR DESCRIPTION
**Pull request by clintonb**
These empty parameters cause errors, resulting in failed responses.

ECOM-5685

### CODE REVIEW

**Comment by clintonb** [comment]
@edx/ecommerce please review

**Comment by clintonb** [comment]
Fixes issues related to #351 and #337.

File: course_discovery/apps/api/filters.py [hunk 1]
```diff
@@ -59,14 +59,29 @@ def build_query(self, **query_filters):
...
+class HaystackRequestFilterMixin:
+    @staticmethod
+    def get_request_filters(request):
+        filters = HaystackFacetFilter.get_request_filters(request)
+
+        # Remove items with empty values.
+        # NOTE: We copy the keys list to avoid a RuntimeError that occurs when modifying a dict
+        # while iterating.
+        for key in list(filters.keys()):
+            if not filters[key]:
```
Comment (MatthewPiatetsky) [inline course_discovery/apps/api/filters.py:76]: What about falsey non-empty values like False?

Comment (clintonb) [reply course_discovery/apps/api/filters.py:76]: `False` would be the string `'False'`. `0` however, is something for which we need to account.

Comment (clintonb) [reply course_discovery/apps/api/filters.py:76]: Test added. Query strings are parsed as...strings. Numbers, including 0, are represented as strings. `'0'` is a non-false value.

**Review by schenedx — APPROVED** [review]
👍

**Comment by rlucioni** [comment]
👍

File: course_discovery/apps/api/filters.py (hunk 1 above)
Comment (MatthewPiatetsky) [reply course_discovery/apps/api/filters.py:76]: if we're making a copy of the keys in memory anyway, another option would be
`filters = [key for key,value in filters.items() if value]`
(or use urllib)
but doesn't matter too much

**Comment by MatthewPiatetsky** [comment]
👍

File: course_discovery/apps/api/filters.py (hunk 1 above)
Comment (clintonb) [reply course_discovery/apps/api/filters.py:76]: :) That's what broke before. I will add a comment in the code explaining why that doesn't work for `QueryDict` objects.

**Pull request merged**